### PR TITLE
Histogram fix

### LINKED ISSUE #8628: [Bug] histogram in graph panel
Please include this information:
- What Grafana version are you using?
4.3.2
- What datasource are you using?
Prometheus
- What OS are you running grafana on?
Ubuntu
- What did you do?
Using the graph panel, wrote a query (that works) and then change the x axis to "Histogram"
- What was the expected result?
To get an histogram with the x axis corresponding to the buckets corresponding the y axis time series.
To get y axis in the histogram view that can be show percentage of counts.
- What happened instead?
x axis values and number of buckets is erratic.
Defining 20 buckets only show 10 buckets
Values are not at all related to the time series y-values ([-20,+20]) while the x axis in the histogram view goes from 0 to 20. Then if i change the number of buckets from auto to 10 i got a x axis range that goes from 0 to 2.0 more or less
- If related to metric query / data viz:

### PR DESCRIPTION
**Pull request by alexanderzobnin**
This PR fixes #8628 bug and adds new option _value_ which is _percent_ by default. So, now histogram should look like [probability density function](https://en.wikipedia.org/wiki/Probability_density_function) for given series (all series are merged into one). Also, I've made some improvements for X axis tick - _short_ values format and 0 tick if series contains both negative and positive values.

### CODE REVIEW

**Comment by daniellee** [comment]
I have a question about percent being default. This means all existing dashboards get percent as default?

File: public/app/plugins/panel/graph/graph.ts [hunk 1]
```diff
@@ -312,10 +312,13 @@ coreModule.directive('grafanaGraph', function($rootScope, timeSrv, popoverSrv) {
               let histMax = _.max(_.map(data, s => s.stats.max));
               let ticks = panel.xaxis.buckets || panelWidth / 50;
               bucketSize = tickStep(histMin, histMax, ticks);
-              let histogram = convertValuesToHistogram(values, bucketSize);
 
+              let normalize = panel.xaxis.histogramValue === 'percent';
```
Comment (alexanderzobnin) [inline public/app/plugins/panel/graph/graph.ts]: @daniellee see here for the default value

**Comment by alexanderzobnin** [comment]
@daniellee If you open Axis editor, default value will be _percent_. But if you don't change it, _count_ will be used (as in the current version). We can change behaviour and use _percent_ as default even it wasn't defined. Not sure which way is better (percent is right value type for histogram, but current is count).

**Comment by torkelo** [comment]
That is normalized by default seems very wrong? And is a breaking change to 4.3 ? Has there been a feature request to normalize the data? Is there a description for which scenarios this preferable and why it is default?

**Comment by torkelo** [comment]
Can we revert this? and only include the fix?

**Comment by alexanderzobnin** [comment]
@daniellee In order to resolve #8761, lets revert it and I'll push fix only. And I'll create another PR for _percent_ feature and discussions.

**Pull request merged**